src/2dplane/ObstacleGrid.cpp

```diff
@@ -24,6 +28,38 @@ Vector2i ObstacleGrid::gridSquareForLocation(const Vector2f &loc) const {
                     loc.y() / height() * discretizedHeight());
 }
 
+float ObstacleGrid::nearestObstacleDist(const Vector2f &state, float maxDist) const {
+    //x and y are the indices of the cell that state is located in
+    float x = (state.x() / (_width / _discretizedWidth));
+    float y = (state.y() / (_height / _discretizedHeight));
+    int xSearchRad = maxDist * _discretizedWidth / (_width * SQRT2);
```

Comment:
My reasoning was that maxDist^2 = xSearchRad^2 and ySearchRad^2, because the maximum possible distance would be the diagonal formed by following the vectors <xSearchRad, 0> + <0, ySearchRad>. I assumed xSearchRad = ySearchRad which meant that maxDist is the diagonal of a square with xSearchRad and ySearchRad as its sides (and the sides are related to the diagonal by a factor of sqrt(2)). On second thought maybe I shouldn't assume xSearchRad = ySearchRad?